### PR TITLE
fix: Update local json with new date format and add verbose logging

### LINKED ISSUE #1689: App crashing in offline state
**Actual Behaviour**

On first time install of the app, the app crashes. This is related to problem in check in date checking in realm repository.

**Expected Behaviour**

App should run fine in offline state.

**Steps to reproduce it**

-

**LogCat for the issue**

-

**Screenshots of the issue**

-

**Would you like to work on the issue?**

-

**Target Milestone to solve the issue**

https://github.com/fossasia/open-event-android/milestone/29

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #1689 

Changes:
- Update local json with assets with updated date format
- Add verbose logging of failures
- Not try to save dates if they are null

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@arp95 @Shailesh351 @shreyasnbhat Please review

File: android/app/src/main/assets/microlocations [hunk 1]
```diff
@@ -1,106 +1,122 @@
 [
-    {
```
Comment (Shailesh351) [inline android/app/src/main/assets/microlocations]: Is this file needs to be changed ?
It is containing date

File: android/app/src/main/assets/sponsors [hunk 2]
```diff
@@ -1,110 +1,155 @@
 [
-    {
```
Comment (Shailesh351) [inline android/app/src/main/assets/sponsors]: Here also ?

File: android/app/src/main/assets/microlocations (hunk 1 above)
Comment (iamareebjamal) [inline android/app/src/main/assets/microlocations]: Yes, but the server response has changed, we don't want to create inconsistent database state

Comment (iamareebjamal) [inline android/app/src/main/assets/microlocations]: When updating event from server, all responses should be covered because they depend on each other

Comment (Shailesh351) [inline android/app/src/main/assets/microlocations]: Ok Great then. Just use `4 space tab` format. https://jsonformatter.org/

Comment (iamareebjamal) [inline android/app/src/main/assets/microlocations]: OK

**Comment by arp95** [comment]
@iamareebjamal working perfect now.

**Comment by iamareebjamal** [comment]
@arp95 Yes, please

**Comment by iamareebjamal** [comment]
Wait. It'll be closed once this is merged

**Comment by arp95** [comment]
@iamareebjamal sure.
can you tell me why i am getting parsing error in my samples. I get a error parsing dates
please have a look here- https://github.com/arp95/open-event/tree/mozilla/sample/MozillaAllHands_17

**Comment by iamareebjamal** [comment]
@arp95 The event time response has changed. Take a look at #1683

**Comment by arp95** [comment]
For this date string: 2017-06-30T10:00:00
it will be like this right: 2017-06-30T10:00:00XXX ?

**Comment by Shailesh351** [comment]
@arp95 No, here  `XXX` is for timezone.

**Comment by arp95** [comment]
@iamareebjamal @Shailesh351 
guys help me here.
The mozilla event is in San Francisco so i kept the timezone field in event.json as "US/Pacific".
And i changed all time fields to this example "2017-06-30T12:00:00-07:00"
On running the app i get the time in utc form. is there something i am doing wrong?

**Comment by iamareebjamal** [comment]
Should be -8:00 and by default you should see the default timing in the app, i.e., 2017-06-30T12:00:00

**Comment by iamareebjamal** [comment]
For example, the API response for Tensoeflow session is 2017-03-18T10:35:00+08:00 for ending time and app correctly shows 1:35 as ending time

**Comment by arp95** [comment]
time zone in san francisco is pst, so i kept timezone field in event field as "US/Pacific" and time strings for example "2017-06-30T08:00:00-07:00"
for above string i am getting time as 15:00. why is it taking utc timezone for event?

**Comment by iamareebjamal** [comment]
It's really strange and it shouldn't

Besides all the tests are working so I think the problem might be in your sample. Perhaps we could take the talk there?

**Comment by arp95** [comment]
sure. doing it.
have a look at he link i posted above and incase of any change in code do tell me over there.

**Comment by iamareebjamal** [comment]
@niranjan94 Please merge

**Pull request merged**